src/main/kotlin/com/bugsnag/android/gradle/BugsnagPlugin.kt

```diff
@@ -170,48 +170,44 @@ class BugsnagPlugin : Plugin<Project> {
             }
 
             // register bugsnag tasks
-            val manifestInfoFile = registerManifestUuidTask(project, variant, output)
-            val proguardTask = when {
+            val manifestInfoFileProvider = registerManifestUuidTask(project, variant, output)
+            val mappingFileProvider = createMappingFileProvider(project, variant, output)
+
+            val proguardTaskProvider = when {
                 jvmMinificationEnabled -> registerProguardUploadTask(
                     project,
-                    variant,
                     output,
                     bugsnag,
-                    httpClientHelperProvider
+                    httpClientHelperProvider,
+                    manifestInfoFileProvider,
+                    proguardUploadClientProvider,
+                    mappingFileProvider
                 )
                 else -> null
-            }
-            val symbolFileTask = when {
+            }?.registerWithAssembleTasks(variant, output, bugsnag, bugsnag.uploadJvmMappings)
+
+            val symbolFileTaskProvider = when {
                 ndkEnabled -> registerSharedObjectUploadTask(
                     project,
-                    variant,
                     output,
                     bugsnag,
-                    httpClientHelperProvider
+                    httpClientHelperProvider,
+                    manifestInfoFileProvider,
+                    ndkUploadClientProvider
                 )
                 else -> null
-            }
-            val releasesTask = registerReleasesUploadTask(project, variant, output, bugsnag)
-
-            // Set the manifest info provider to prevent reading
-            // the manifest more than once
-            proguardTask?.get()?.let { task ->
-                task.manifestInfoFile.set(manifestInfoFile)
-                val mappingFileProvider = createMappingFileProvider(project, variant, output)
-                task.mappingFileProperty.set(mappingFileProvider)
-                releasesTask.get().jvmMappingFileProperty.set(mappingFileProvider)
-                task.uploadRequestClient.set(proguardUploadClientProvider)
-            }
-
-            symbolFileTask?.get()?.let { task ->
-                val ndkSearchDirs = symbolFileTask.get().searchDirectories
-                releasesTask.get().ndkMappingFileProperty.from(ndkSearchDirs)
-                task.uploadRequestClient.set(ndkUploadClientProvider)
-            }
-            releasesTask.get().uploadRequestClient.set(releasesUploadClientProvider)
-            releasesTask.get().manifestInfoFile.set(manifestInfoFile)
-            symbolFileTask?.get()?.manifestInfoFile?.set(manifestInfoFile)
-            releasesTask.get().manifestInfoFile.set(manifestInfoFile)
+            }?.registerWithAssembleTasks(variant, output, bugsnag, project.provider { true })
```

Comment:
I think this should probably be `project.provider { bugsnag.uploadNdkMappings }`

Reply:
Good call. It was hardcoded to `true` before so I was just matching this originally